Sources/RxMoya/RxMoyaProvider.swift

```diff
@@ -69,3 +74,10 @@ public extension RxMoyaProvider {
         }
     }
 }
+
+public extension MoyaProvider {
+    
+    public var rx: RxMoyaProvider<Target> {
```

Comment:
I think we just need to conform to `ReactiveCompatible` like:
```swift
extension MoyaProvider: ReactiveCompatible {}
```

And the same for ReactiveSwift. This way we get `.rx` & `.reactive` properties for free and we could remove `RxProvider` & `ReactiveProvider`, using only `rx.request` & `reactive.request` (which we have already implemented, just need to extract request functions from providers into extensions to `Reactive` structs)